src/consoleappender.cxx

```diff
@@ -47,27 +48,54 @@ ConsoleAppender::getOutputMutex ()
     return helpers::getConsoleOutputMutex ();
 }
 
+static
+std::locale
+get_locale_by_name(tstring const & locale_name)
```

Comment:
This function is clearly a copy of the same function from `fileappender.cxx`. Instead of duplicating the code, please move the function definition into `env.cxx` and declaration into `internal/env.h`.